homeassistant/components/glances/sensor.py

```diff
@@ -328,6 +328,18 @@ def __init__(
         )
         self._attr_unique_id = f"{coordinator.config_entry.entry_id}-{sensor_name_prefix}-{description.key}"
 
+    @property
+    def available(self) -> bool:
+        """Set sensor unavailable when native value is invalid."""
+        if super().available:
+            return (
+                not self._numeric_state_expected
+                or isinstance(self.native_value, (int, float))
```

Comment:
Every time we access `self.native_value` there's another `isinstance` check besides some dict lookups. I suggest storing the result the first time in a local variable for reuse.

Reply:
Sounds like a good improvement, but the PR is already merged, so I am going to add it in another PR.
Dict lookups are not expensive at all with O(1) compared to the isinstance call. I think this can be neglected. I am not aware of the impact and performance hit of isinstance, but I agree that generally faster is better!